### DISCUSSION TITLE
Waiting until the image is created

### DISCUSSION THREAD

**Discussion by psads-git** [typst/typst#4576] [Q&A] [answered]
Consider the following line of code:
```
#image("sin.svg")
```

While the image file is not created, `typst` keeps throwing out a compiling error, which is entirely justified. I am wondering whether is it possible avoiding such compiling errors. I am thinking about something like `try` and `exception` of Python.

**Comment by jamesrswift** [comment]
![image](https://github.com/user-attachments/assets/7bd5b643-ffb9-4413-8f3d-13fd3f6032a5)

**Comment by freundTech** [comment, marked as answer]
There is currently no functionality in typst to allow this, but there is a hacky workaround to replace missing images with placeholders: https://sitandr.github.io/typst-examples-book/book/typstonomicon/try_catch.html

**Comment by psads-git** [comment]
Very nice solution, @freundTech! Thanks!

**Comment by psads-git** [comment]
Let me add that, meanwhile, I have come across a very nice package that does that automatically: [`prequery`](https://typst.app/universe/package/prequery/).
